share/shwap/p2p/bitswap/getter.go

```diff
@@ -230,17 +258,37 @@ func (g *Getter) GetNamespaceData(
 	return nsShrs, nil
 }
 
-// session decides which fetching session to use for the given header.
-func (g *Getter) session(ctx context.Context, hdr *header.ExtendedHeader) exchange.Fetcher {
-	session := g.archivalSession
+// isArchival reports whether the header is for archival data
+func (g *Getter) isArchival(hdr *header.ExtendedHeader) bool {
+	return !availability.IsWithinWindow(hdr.Time(), g.availWndw)
+}
 
-	isWithinAvailability := availability.IsWithinWindow(hdr.Time(), g.availWndw)
-	if isWithinAvailability {
-		session = g.availableSession
+// session takes a session out of the respective session pool
+func (g *Getter) session(isArchival bool) exchange.Fetcher {
+	if isArchival {
+		v := g.archivalPool.Get()
+		if v == nil {
+			panic("Getter must be started")
+		}
+
+		return v.(exchange.Fetcher)
 	}
 
-	trace.SpanFromContext(ctx).SetAttributes(attribute.Bool("within_availability", isWithinAvailability))
-	return session
+	v := g.availablePool.Get()
+	if v == nil {
+		panic("Getter must be started")
+	}
+
+	return v.(exchange.Fetcher)
+}
+
+// poolSession puts session back into the respective session pool
+func (g *Getter) poolSession(session exchange.Fetcher, isArchival bool) {
```

Comment:
Nit: more common naming will be `putSession`.

Reply:
Sure